### ISSUE TITLE
get_installed_language_packs gives very different output from one call to the next

### ISSUE THREAD

**Issue by rtibbles** [learningequality/ka-lite#3311]
Branch: develop

Expected behaviour: Unless a new language pack has been installed, the output of get_installed_language_packs should be identical from call to call

Current behaviour: Repeated calls to the function result in seemingly radically different outputs.

Steps to reproduce:
Inject a logging.debug statement inside 'get_best_available_language' and log the output of the call to get_installed_language_packs.

**Comment by aronasorman** [comment]
Strange, I consistently get english here (despite having more languages installed):

```
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
DEBUG:kalite:['en']
DEBUG:kalite:Requested code eo, got code en
```

**Comment by rtibbles** [comment]
Not quite as weird as getting Russian and Xhosa, but still not ideal.

**Comment by aronasorman** [comment]
I stepped through a single call of `get_installed_language_packs` and it seems to be doing the right thing.

However it looks like `select_best_available_language` is getting its list of languages differently, i.e. calculated by getting the list of languages with khan-exercises: https://github.com/learningequality/ka-lite/blob/develop/kalite/topic_tools/__init__.py#L143-L146

**Comment by aronasorman** [comment]
So I'm not sure if that's the correct way of doing things anymore, as we could have languages translated in Perseus but not in khan-exercises (yet).

**Comment by rtibbles** [comment]
Ah yes. Problematico. We need to make that more flexible.

**Comment by rtibbles** [comment]
Can do a check conditionally on exercise.get("uses_assessment_items")

**Comment by aronasorman** [comment]
Gotcha. I'll open a PR soon.

**Comment by aronasorman** [comment]
Managed to replicate the changing language codes as well:
```
DEBUG:kalite:choosing best language among ['en', 'pl']
DEBUG:kalite:choosing best language among ['ru', 'en']
DEBUG:kalite:choosing best language among ['ru', 'fr', 'en', 'pt', 'de', 'tr', 'ja', 'es']
DEBUG:kalite:choosing best language among ['ru', 'en', 'pt', 'pt-PT', 'tr', 'es']
DEBUG:kalite:choosing best language among ['en', 'ar']
DEBUG:kalite:choosing best language among ['ru', 'en', 'mn', 'da', 'ar', 'es', 'pl']
DEBUG:kalite:choosing best language among ['ru', 'en', 'ar', 'es']
DEBUG:kalite:choosing best language among ['mn', 'ru', 'en', 'pl', 'da']
DEBUG:kalite:choosing best language among ['fr', 'en', 'es']
DEBUG:kalite:choosing best language among ['ru', 'xh', 'en', 'pt', 'pt-PT', 'tr', 'fr', 'ur', 'ar', 'zul', 'es']
DEBUG:kalite:choosing best language among ['ru', 'xh', 'en', 'pt', 'pt-PT', 'fr', 'da', 'pl', 'ja', 'es']
DEBUG:kalite:choosing best language among ['ru', 'en', 'mn', 'da', 'es', 'pl']
DEBUG:kalite:choosing best language among ['el', 'en', 'pt', 'pt-PT', 'mn', 'tr', 'da', 'pl', 'es']
DEBUG:kalite:choosing best language among ['en', 'tr']
DEBUG:kalite:choosing best language among ['en']
DEBUG:kalite:choosing best language among ['en', 'pl', 'da']
DEBUG:kalite:choosing best language among ['ru', 'xh', 'en', 'zh', 'pt', 'pt-PT', 'pa', 'tr', 'fa', 'ar', 'ja', 'es']
DEBUG:kalite:choosing best language among ['ru', 'fr', 'en', 'pt', 'sw', 'bn', 'de', 'tr', 'ur', 'ar', 'es', 'he']
DEBUG:kalite:choosing best language among ['ru', 'fr', 'en', 'es']
DEBUG:kalite:choosing best language among ['zul', 'en', 'pt', 'pt-PT', 'bn', 'sw', 'it', 'ur', 'ar', 'pl', 'th', 'es', 'he']
DEBUG:kalite:choosing best language among ['ru', 'fr', 'en', 'pt', 'pt-PT', 'tr', 'da', 'pl', 'ur', 'ja', 'es']
DEBUG:kalite:choosing best language among ['ru', 'en', 'ar', 'es', 'tr']
```

**Comment by aronasorman** [comment]
Strange, as I really don't have any of these languages installed at all.

**Comment by rtibbles** [comment]
Yeah, that was what I was getting.

**Comment by aronasorman** [comment]
It's coming from a call to https://github.com/learningequality/ka-lite/blob/develop/kalite/topic_tools/__init__.py#L255. That's the list of dubbed languages for a certain content.

**Comment by aronasorman** [comment]
```
(Pdb) dubmap
{'en': 'C-2Ln0pK3kY', 'pt': '3LjK9pniX-4', 'pt-PT': '3LjK9pniX-4', 'tr': 'f13rkhOdlbQ', 'da': 'Fv1-e1CxnDU', 'ar': 'QJV7JI7OL2s', 'pl': 'GxRqvdQuOko', 'es': 'ZEwEWXFYIoY'}
```

**Comment by aronasorman** [comment]
I think this is the proper behaviour in this case.

**Comment by aronasorman** [comment]
One of the bugs uncovered by this issue has been fixed in #3328. The original issue turns out to be expected behaviour. Closing.
